secretsmanager/src/main/scala/uk/gov/nationalarchives/DASecretsManagerClient.scala

```diff
@@ -0,0 +1,174 @@
+  }
+
+  /** Describes a secret
+    * @return
+    *   A DescribeSecretResponse object wrapped in F[_]
+    */
+  def describeSecret(): F[DescribeSecretResponse] = {
+    val request = DescribeSecretRequest.builder().secretId(secretId).build()
+    secretsManagerAsyncClient.describeSecret(request).liftF
+  }
+
+  /** Gets a secret value by stage with a default of Current
+    * @param stage
+    *   The stage you want the value for, either Current or Pending
+    * @param decoder
+    *   A circe decoder to convert the secrets manager response to a case class
+    * @tparam T
+    *   The type of the case class to return
+    * @return
+    *   An instance of type T wrapped in F[_]
+    */
+  def getSecretValue[T](stage: Stage = Current)(implicit decoder: Decoder[T]): F[T] = secretValue(stage)
+
+  /** Gets a secret value by stage and version id
+    * @param stage
+    *   The stage you want the value for, either Current or Pending
+    * @param versionId
+    *   The version id of the secret you want to retrieve.
+    * @param decoder
+    *   A circe decoder to convert the secrets manager response to a case class
+    * @tparam T
+    *   The type of the case class to return
+    * @return
+    *   An instance of type T wrapped in F[_]
+    */
+  def getSecretValue[T](versionId: String, stage: Stage)(implicit decoder: Decoder[T]): F[T] =
+    secretValue(stage, Option(versionId))
+
+  /** Creates a new secret value for a given stage and optional request token.
+    * @param secret
+    *   An object of type T. This will be deserialised to a string and passed to secrets manager
+    * @param stage
+    *   The stage to put the value into
+    * @param clientRequestToken
+    *   An optional clientRequestToken
+    * @param encoder
+    *   A circe encoder to convert the secret from T to String
+    * @tparam T
+    *   The type of the secret object
+    * @return
+    *   PutSecretValueResponse wrapped in F[_]
+    */
+  def putSecretValue[T](secret: T, stage: Stage = Current, clientRequestToken: Option[String] = None)(implicit
+      encoder: Encoder[T]
+  ): F[PutSecretValueResponse] = {
+    val builder = PutSecretValueRequest.builder
+      .secretId(secretId)
+      .secretString(secret.asJson.printWith(noSpaces))
+      .versionStages(stage.toString)
+    val request = clientRequestToken.map(builder.clientRequestToken).getOrElse(builder).build
+    secretsManagerAsyncClient.putSecretValue(request).liftF
+  }
+
+  /** Update the version stage. This is used to finalise secret rotation.
+    * @param moveToVersionId
+    *   The version you are moving to
+    * @param removeFromVersionId
+    *   The version you are moving from
+    * @param stage
+    *   The stage of the version
+    * @return
+    *   UpdateSecretVersionStageResponse wrapped in F[_]
+    */
+  def updateSecretVersionStage(
+      moveToVersionId: String,
+      removeFromVersionId: String,
```

Comment:
Same with this, it's what AWS use so we should stick with that if we can.

Reply:
It's weird because it's the name of their method, which makes sense, but also the name that they gave the parameter of this method but in their documentation they write "The ID of the version that the staging label is to be removed from."